src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -99,7 +99,11 @@ function runEventInBatch(event) {
 }
 
 function getInstIfValueChanged(targetInst) {
-  if (inputValueTracking.updateValueIfChanged(targetInst)) {
+  if (
+    inputValueTracking.updateValueIfChanged(
+      ReactDOMComponentTree.getNodeFromInstance(targetInst),
```

Comment:
Minor nit but it might look a bit cleaner if you extract inner expression into `const targetNode = ...`